services/src/main/java/org/fao/geonet/api/records/formatters/PdfOrHtmlResponseWriter.java

```diff
@@ -0,0 +1,53 @@
+package org.fao.geonet.api.records.formatters;
+
+import jeeves.server.context.ServiceContext;
+import org.fao.geonet.Constants;
+import org.fao.geonet.api.records.extent.MapRenderer;
+import org.fao.geonet.constants.Geonet;
+import org.fao.geonet.kernel.setting.SettingManager;
+import org.fao.geonet.util.XslUtil;
+import org.fao.geonet.utils.Log;
+import org.springframework.stereotype.Component;
+import org.xhtmlrenderer.pdf.ITextRenderer;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@Component
+class PdfOrHtmlResponseWriter {
+
+    void writeOutResponse(ServiceContext context, String metadataUuid, String lang, HttpServletResponse response, FormatType formatType, byte[] formattedMetadata) throws Exception {
+        response.setContentType(formatType.contentType);
+        String filename = "metadata-" + metadataUuid + "." + formatType;
+        response.addHeader("Content-Disposition", "inline; filename=\"" + filename + "\"");
```

Comment:
This code was moved not implemented as part of the PR. Since the given metadatauuid and formatType are checked to comply with database uuids and enum precondition, these failing checks are not valid security vulnerabilities.